test/blackbox-tests/test-cases/pkg/pin-depends.t

```diff
@@ -175,10 +175,17 @@ Pin to an invalid opam file
   unsupported or missing file format version; should be 2.0 or older
   [1]
 
-Pin to an HTTP archive doesn't work
+Pin to an HTTP archive work
 
-  $ runtest "http://0.0.0.0/tarball.tgz"
-  File "foo.opam", line 1, characters 0-0:
-  Error: Could not determine location of repository http://0.0.0.0/tarball.tgz
-  Hint: Specify either a file path or git repo via SSH/HTTPS
-  [1]
+  $ mkdir _source/
+  $ cat > _source/bar.opam << EOF
+  > opam-version: "2.0"
+  > EOF
+  $ tar cf tarball.tar -C _source bar.opam
+  $ MD5_CHECKSUM=$(md5sum tarball.tar  | cut -f1 -d' ')
+  $ echo tarball.tar > fake-curls
+  $ PORT=1
+  $ runtest "http://0.0.0.0:$PORT/tarball.tar" > output
+  Solution for dune.lock:
+  - bar.1.0.0
+  $ cat output | grep "md5=$MD5_CHECKSUM" 2>&1 > /dev/null
```

Comment:
Nitpick but useless use of `cat`. Besides that I think this is a very good solution.